private/user.js

```diff
@@ -0,0 +1,86 @@
+const router = require('express').Router();
+const db = require('../database/connection.js').connection;
+const bcrypt = require('bcrypt');
+
+// import utils class
+const Utilities = require('./models/Utils').Utils;
+// create utils object
+const Utils = new Utilities();
+
+// the cost factor that indicates the amount of time needed to calculate a single bcrypt hash.
+// Higher the salt rounds, the more hashing rounds are done,
+// hence the time and difficulty is increased while brute-forcing
+const saltRounds = 15;
+
+//Dagmara
+//check if username and password exists in db and return player id if not return message 
+router.post('/api/user/login', (req, res) => {
+  db.query('SELECT * FROM player WHERE username=?;', [req.body.username], (error, result, fields) => {
+    if (result && result.length === 1) {
+      bcrypt.compare(req.body.password, result[0].password, (error, match) => {
+        if (match) {
+          res.send({
+            playerId: result[0].player_id,
+            username: result[0].username,
+          });
+        } else {
```

Comment:
As this else and the one below are the same can we maybe just do one? Instead of else just put it at the end to do it by default? Or if it has problem that it will call both res.send for correct data and for incorrect you can just put "return" after the correct one?

Reply:
I can't do it bc it doesn't recognise the headers
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client